main/src/ca/uwaterloo/flix/language/ast/TypedAst.scala

```diff
@@ -215,6 +215,8 @@ object TypedAst {
 
     case class TryWith(exp: Expr, effUse: EffectSymUse, rules: List[HandlerRule], tpe: Type, eff: Type, loc: SourceLocation) extends Expr
 
+    case class RunWith(exp: Expr, handler: Expr, tpe: Type, eff: Type, loc: SourceLocation) extends Expr
```

Comment:
I would suggest:

```
TryWith -> RunWith (can be separate PR)
RunWith -> RunWithHandler (this PR)
```

Reply:
You should write `exp1` and `exp2` everywhere. (Thats what we do right? :))

Reply:
yeah i just don't like it, but I'll change it 😄